projects/kit/components/select/native-select/native-select.component.ts

```diff
@@ -0,0 +1,89 @@
+import {DOCUMENT, NgForOf, NgIf, NgTemplateOutlet} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    inject,
+    Input,
+    signal,
+} from '@angular/core';
+import {tuiAsControl, TuiControl} from '@taiga-ui/cdk/classes';
+import type {TuiBooleanHandler} from '@taiga-ui/cdk/types';
+import {tuiInjectElement} from '@taiga-ui/cdk/utils/dom';
+import type {TuiTextfieldAccessor} from '@taiga-ui/core/components/textfield';
+import {
+    tuiAsTextfieldAccessor,
+    TuiTextfieldComponent,
+    TuiWithTextfield,
+} from '@taiga-ui/core/components/textfield';
+import {TUI_ITEMS_HANDLERS} from '@taiga-ui/core/tokens';
+import {tuiIsFlat} from '@taiga-ui/kit/utils';
+
+@Component({
+    standalone: true,
+    selector: 'select[tuiSelect]',
+    imports: [NgForOf, NgIf, NgTemplateOutlet],
+    templateUrl: './native-select.template.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [tuiAsTextfieldAccessor(TuiNativeSelect), tuiAsControl(TuiNativeSelect)],
+    hostDirectives: [TuiWithTextfield],
+    host: {
+        '[attr.aria-label]': 'ariaLabel',
+        '[attr.aria-invalid]': 'invalid()',
+        '[disabled]': '!interactive()',
+        '[value]': 'stringified()',
+        '(change)': 'selectOption($event.target.options.selectedIndex)',
+    },
+})
+export class TuiNativeSelect<T>
+    extends TuiControl<T | null>
+    implements TuiTextfieldAccessor<T>
+{
+    private readonly doc = inject(DOCUMENT);
+    private readonly element = tuiInjectElement<HTMLSelectElement>();
+
+    protected readonly isFlat = tuiIsFlat;
+    protected readonly placeholder = signal('');
+    protected readonly itemsHandlers = inject(TUI_ITEMS_HANDLERS);
+    protected readonly textfield: TuiTextfieldComponent<T> =
+        inject(TuiTextfieldComponent);
+
+    protected readonly stringified = computed((value = this.value()) =>
+        value ? this.textfield.stringify(value) : '',
+    );
+
+    protected readonly showPlaceholder = computed(
+        () => this.placeholder() && !this.stringified(),
+    );
+
+    @Input()
+    public items: ReadonlyArray<readonly T[]> | readonly T[] | null = [];
+
+    @Input()
+    public labels: readonly string[] = [];
+
+    @Input()
+    public disabledItemHandler!: TuiBooleanHandler<T> | null;
+
+    // TODO(v5): use signal inputs
+    @Input('placeholder')
+    public set placeholderSetter(x: string) {
+        this.placeholder.set(x);
+    }
+
+    public setValue(value: T | null): void {
+        this.onChange(value);
+    }
+
+    protected get ariaLabel(): string | null {
+        return this.doc.querySelector(`label[for="${this.element.id}"]`)
```

Comment:
That's heavy to do on every change detection cycle 🤔 Maybe it's not worth it?

Reply:
Honestly, I don't know 😄 
It is just copy-pasted solution from previous generation of select
https://github.com/taiga-family/taiga-ui/blob/53345270f2322d287c9318d2b09589b98664840f/projects/core/components/textfield/select.directive.ts#L53-L56

I think we can ask @vladimirpotekhin about it tomorrow
- https://github.com/taiga-family/taiga-ui/pull/10362

Reply:
Let's drop it for now. If it's that computationally taxing to automate that, people can always set it themselves if they need it.

Reply:
Dropped in https://github.com/taiga-family/taiga-ui/pull/10642/commits/a7f2f24e0b696a69f32a33196649cd808225a9d7